File: tests/fixtures/polyglot/ts/audit/audit.test.ts
import { AuditLog } from "./audit";
import { TaxCalculator } from "../billing/tax";

export function testReviewTotal(): void {
  const log = new AuditLog();
  const total = log.reviewTotal(100);
  if (total !== 120) throw new Error(`expected 120, got ${total}`);
}

export function testTaxRate(): void {
  const calc = new TaxCalculator(0.1);
  if (calc.taxFor(50) !== 5) throw new Error("bad tax");
}

File: tests/fixtures/polyglot/go/audit/audit_test.go
package audit

import (
	"testing"

	"polyglot/billing"
)

func TestReviewTotal(t *testing.T) {
	log := NewAuditLog()
	total := log.ReviewTotal(100.0)
	if total != 120.0 {
		t.Fatalf("expected 120.0, got %f", total)
	}
}

func TestRound2(t *testing.T) {
	if billing.Round2(1.005) != 1.01 {
		t.Fatal("bad rounding")
	}
}
